src/software/thunderscope/robot_diagnostics/motor_fault_view.py

```diff
@@ -0,0 +1,179 @@
+        """
+        for key in self.motor_faults.keys():
+            self.enabled = self.enabled and self.motor_faults[key].enabled
+
+            motor_faults = self.motor_faults[key].motor_faults
+
+            byte_array = QByteArray()
+            buffer = QBuffer(byte_array)
+            buffer.open(QIODeviceBase.OpenModeFlag.ReadWrite)
+
+            if self.enabled and motor_faults:
+                get_warning_icon().save(buffer, "PNG")
+            elif motor_faults:
+                get_stopped_icon().save(buffer, "PNG")
+            else:
+                get_no_fault_icon().save(buffer, "PNG")
+
+            image_data = byte_array.toBase64()
+
+            self.fault_count += len(self.motor_faults[key].motor_faults)
+
+            self.motor_fault_tooltip += (
+                "<p>"
+                "<img src='data:image/png;base64, "
+                f"{str(image_data.data(), encoding='utf-8')}'"
+                'width="10" height="10"/>'
+                "&nbsp;"
+                f"<b>{key}</b>"
+            )
+
+            if motor_faults:
+                self.motor_fault_tooltip += (
+                    "<ul>"
+                    + "".join(
+                        f"<li><p>{get_string_val(enum_descriptor, fault)}</p></li>"
+                        for fault in motor_faults
+                    )
+                    + "</ul>"
+                )
+            else:
+                self.motor_fault_tooltip += f"<ul><li>No Motor Faults :)</li></ul>"
+
+    def update_ui(self):
+        """
+        Updates the main UI according to the current motor states
+
+        If the widget has not received a signal yet (self.enabled = None),
+        sets the UI to the default uninitialized state
+
+        Else If any motor has a fault,
+        - the notification label is visible and displays the current number of faults
+        - the color and text of the main display changes
+            - if all motors are still enabled, color is yellow and text is "Warning"
+            - if any motor is disabled, color is red and text is "Error"
+        """
+        if self.enabled is None:
+            self.motor_fault_display.setStyleSheet("background: grey; color: white;")
+            self.motor_fault_display.setText("No Signal")
+        else:
+            if self.fault_count:
+                self.fault_count_label.setText(str(self.fault_count))
+                self.fault_count_label.show()
+            else:
+                self.fault_count_label.hide()
+
+            self.motor_fault_display.setStyleSheet("background: green;")
+            self.motor_fault_display.setText("No Fault")
+
+            if self.fault_count:
+                self.motor_fault_display.setStyleSheet(
+                    "background: yellow; color: black;"
+                    if self.enabled
+                    else "background: red;"
+                )
+                self.motor_fault_display.setText("Warning" if self.enabled else "Error")
```

Comment:
nit: the logic here can be slightly simplified into a single if else statement
```suggestion
            if self.fault_count > 0:
                self.fault_count_label.setText(str(self.fault_count))
                self.fault_count_label.show()
                
                self.motor_fault_display.setStyleSheet(
                    "background: yellow; color: black;"
                    if self.enabled
                    else "background: red;"
                )
                self.motor_fault_display.setText("Warning" if self.enabled else "Error")
            else:
                self.fault_count_label.hide()

                self.motor_fault_display.setStyleSheet("background: green;")
                self.motor_fault_display.setText("No Fault")
```